sleap/nn/tracking.py

```diff
@@ -443,7 +576,11 @@ def _init_matching_queue(self):
         return deque(maxlen=self.track_window)
 
     def reset_candidates(self):
-        self.track_matching_queue = deque(maxlen=self.track_window)
+        if self.max_tracking and self.track_matching_queue_dict:
```

Comment:
Do we need to check for `self.track_matching_queue_dict`? This will be initialized by the factory method in the attribute definition anyway.